File: example/increment.py
from pyspark.sql import SparkSession


if __name__ == "__main__":
    spark = SparkSession\
        .builder\
        .getOrCreate()

    sc = spark.sparkContext


    dataRdd = sc.textFile("/home/student/css333/dataset/ones.txt",2)

    newRdd = dataRdd.map(lambda str: int(str) + 1)

    sum = newRdd.reduce(lambda i,j: i+j)
    print("Summation of all numbers = %d" % sum)

    print("Save newRdd to file")
    newRdd.saveAsTextFile("/home/student/css333/example/newRDD")

    spark.stop()

File: example/Increment.java
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.SparkSession;

public class Increment {
   public static void main(String[] args) {
      SparkSession spark = SparkSession
            .builder()
            .getOrCreate();
      JavaSparkContext jsc = new JavaSparkContext(spark.sparkContext());

      
      JavaRDD<String> dataRdd = jsc.textFile("/home/student/dataset/ones.txt",2);

      JavaRDD<Integer> newRdd = dataRdd.map( str -> 
            Integer.parseInt(str) + 1
      ); 

      Integer sum = newRdd.reduce( (i,j) -> i+j );
      System.out.println("Summation of all numbers = " + sum);
   
      System.out.println("Save newRdd to file");
      newRdd.saveAsTextFile("/home/student/output");

      spark.stop();
   }

}
